### PR TITLE
Add image encoder base

### PR DESCRIPTION
**Pull request by EdenWuyifan**

### CODE REVIEW

File: tests/test_data/image_dataset_2/tables/learningData.csv
Comment (roquelopez) [inline tests/test_data/image_dataset_2/tables/learningData.csv:1]: Remove the d3mIndex column. Avoid to use the name `learningData.csv`. Use something more descriptive, e.g. digits.csv, etc. It's not necessary to have the folder `tables`. Change the name `image_dataset_2` to `digits`

File: examples/adding_clip_primitives.ipynb
Comment (roquelopez) [inline examples/adding_clip_primitives.ipynb:1]: Add only one example for image classification. Add the link to download the dataset in the jupyter notebook. This link should contain the trian_data.csv and test_data.csv files (use the same names that we have in the other example to be consistent). Avoid to split the data in the jupyter notebook. The less code we put in our example, the better.

**Review by roquelopez — CHANGES_REQUESTED** [review]
In general it looks good, thanks! Please address my comments!

**Comment by roquelopez** [comment]
I made some minor changes. Feel free to merge it.

**Comment by EdenWuyifan** [comment]
I fixed some bugs and new it is ready to merge

**Pull request merged**